plugins/puppetdb/puppetdb

```diff
@@ -0,0 +1,122 @@
+#!/usr/bin/python3
+#
+# This code is licensed under GPLv3+
+#
+# Create a graph out of PuppetDB's JVM memory usage.
+#
+# dependencies: python3-requests
+#
+# compatibility:
+#  * PuppetDB 6.x: https://puppet.com/docs/puppetdb/6.0/api/metrics/v1/mbeans.html#jvm-metrics
+#
+# Configuration can be done via environment variables, usually via a munin
+# configuration file for this plugin (e.g.
+# /etc/munin/plugin-conf.d/puppetdb.conf)
+#
+#  * pdburl -- Set the URL to your PuppetDB instance. This url should point to
+#              the mbeans endpoint. By default this has a value of
+#              http://localhost:8080/metrics/v1/mbeans
+#
+#  * timeout -- Time in seconds (int) to wait for a result when querying the REST
+#               API. By default, wait for 2 seconds
+#
+import os
+import sys
+import requests
+
+class WrongStatusCode(Exception):
+    pass
+
+
+def rest_request(url, timeout):
+    """Make a GET request to URL. We expect a 200 response.
+    This function will let exception from requests raise through to indicate request failure.
+    If response code is not 200, it will raise a WrongStatusCode exception.
+    """
+    headers = {'content-type': 'application/json', 'Accept-Charset': 'UTF-8'}
+    resp = requests.get(url, headers=headers, timeout=timeout)
+    if resp.status_code != 200:
+        raise WrongStatusCode(f"GET Request to 'url' returned code {resp.status_code}. expected 200.")
+    return resp
+
+
+def config():
+    """Print all graph configuration for munin."""
+    print("graph_title PuppetDB JVM Memory usage")
+    print("graph_args --base 1024") # TODO not sure about unit
+    print("graph_vlabel Bytes")
+    print("graph_info This graph shows how much memory from the JVM Heapspace is being used by PuppetDB")
+    # This is super weird, but currently puppet graphs are in this
+    # category so let's place puppetdb graphs in the same category
+    print("graph_category devel")
```

Comment:
I think, you should use the proper category here. Feel free to correct the category of the other `puppet` plugin.

Reply:
I took a look a the "well-known categories" (http://guide.munin-monitoring.org/en/latest/reference/graph-category.html#well-known-categories) and couldn't find something that would fit better.

should we create a category like "configuration_management" or should they rather be in a "puppet" category ?

Reply:
I would suggest `system` or `other` for now. The list of categories is surely showing its age, but let us handle this separately.